src/gui/wallet.go

```diff
@@ -28,6 +28,98 @@ type UnconfirmedTxnsResponse struct {
 	Transactions []visor.ReadableUnconfirmedTxn `json:"transactions"`
 }
 
+type walletEntry struct {
+	Address string `json:"address"`
+	Public  string `json:"public_key"`
+}
+
+type walletMeta struct {
+	Coin       string `json:"coin"`
+	Filename   string `json:"filename"`
+	Label      string `json:"label"`
+	Type       string `json:"type"`
+	Version    string `json:"version"`
+	CryptoType string `json:"cryptoType"`
+	Timestamp  int64  `json:"timestamp"`
+	Encrypted  bool   `json:"encrypted"`
+}
+
+// WalletResponse wallet response struct for http apis
+type WalletResponse struct {
+	Meta    walletMeta    `json:"meta"`
+	Entries []walletEntry `json:"entries"`
+}
+
+// NewWalletResponse creates WalletResponse struct from *wallet.Wallet
+func NewWalletResponse(w *wallet.Wallet) (*WalletResponse, error) {
+	var wr WalletResponse
+
+	wr.Meta.Coin = w.Meta["coin"]
+	wr.Meta.Filename = w.Meta["filename"]
+	wr.Meta.Label = w.Meta["label"]
+	wr.Meta.Type = w.Meta["type"]
+	wr.Meta.Version = w.Meta["version"]
+	wr.Meta.CryptoType = w.Meta["cryptoType"]
+
+	// Converts "encrypted" string to boolean if any
+	if encryptedStr, ok := w.Meta["encrypted"]; ok {
+		encrypted, err := strconv.ParseBool(encryptedStr)
+		if err != nil {
+			return nil, err
+		}
+		wr.Meta.Encrypted = encrypted
+	}
+
+	if tmStr, ok := w.Meta["tm"]; ok {
+		// Converts "tm" string to integer timestamp.
+		tm, err := strconv.ParseInt(tmStr, 10, 64)
+		if err != nil {
+			return nil, err
+		}
+		wr.Meta.Timestamp = tm
+	}
+
+	for _, e := range w.Entries {
+		wr.Entries = append(wr.Entries, walletEntry{
+			Address: e.Address.String(),
+			Public:  e.Public.Hex(),
+		})
+	}
+
+	return &wr, nil
+}
+
+// ToWallet converts WalletResponse to wallet.Wallet
+func (wr *WalletResponse) ToWallet() (*wallet.Wallet, error) {
```

Comment:
Do we need this? Can gui.Client just return *WalletResponse?

Reply:
Hmm, maybe can let gui.Client return *WalletResponse, and move the convert function into integration_test.go, as the tests are using *wallet.Wallet.

Reply:
ok, do that